pkg/pillar/cmd/verifier/verifier.go

```diff
@@ -541,7 +373,11 @@ func handleCreate(ctx *verifierContext, objType string,
 		return
 	}
 
-	ok, size := markObjectAsVerifying(ctx, config, &status)
+	// We generate a temporary UUID to avoid conflicts
+	// where multiple different objects can have a different claimed sha256
+	// Of course, only one of those will pass the verification.
+	tmpID := uuid.NewV4()
```

Comment:
Just curious, in which scenario we will have conflicts?

Reply:
Claimed sha collision.

It is an obscure case but if a device is asked to download contentA with claimed sha (in the ContentInfo from the controller) being X, and also contentB with the same claimed sha X, then it will end up with two separate files targeting the same filename. Of course, only one of A and B will have sha X, but we don't know that until the verifer has completed and we know that the claimed sha is correct.

Up until yesterday's pull that was handled by having an ImageID as part of the verifier filename. Note the sha has always been the filename once the verification has completed.

Reply:
So the file during verification will be named `verifier/<nonce>-<sha>` (or in this case `verifier/<imageID>-<sha>`), but once verified, it will just be `verified/<sha>`? That makes sense.

Reply:
Correct